consumer/offset_store.go

```diff
@@ -277,14 +278,16 @@ func (r *remoteBrokerOffsetStore) read(mq *primitive.MessageQueue, t readType) i
 	r.mutex.RLock()
 	switch t {
 	case _ReadFromMemory, _ReadMemoryThenStore:
-		defer r.mutex.RUnlock()
 		off, exist := r.OffsetTable[*mq]
 		if exist {
+			r.mutex.RUnlock()
```

Comment:
could you replace all `RUnlock` with `defer`?

Reply:
`defer` wound `RUnlock ` an Unlocked  mutex at the end of case block.

Reply:
I mean could use defer at Line 239 that next line of `r.mutex.RLock()`, but I found there is a lock-unlock pair in the update function, so keep it now may better